### PR TITLE
[bgpd] Check zebra is ready to connect when starting bgpd

### PR DESCRIPTION
**Pull request by shi-su**
**- Why I did it**
There is a race condition between zebra server accepts connections and bgpd tries to connect. Bgpd has a chance to try to connect before zebra is ready. In this scenario, bgpd will try again after 10 seconds and operate as normal within these 10 seconds. As a consequence, whatever bgpd tries to sent to zebra will be missing in the 10 seconds. To avoid such a scenario, bgpd should start after zebra is ready to accept connections.

Fix https://github.com/Azure/sonic-buildimage/issues/5026

**- How I did it**
Check and wait until zebra is ready to accept connections before starting bgpd.

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: dockers/docker-fpm-frr/bgpd.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+while getopts ":A:P:" opt; do
+    case "${opt}" in
+        A) addr=${OPTARG}
+            ;;
+        P) port=${OPTARG}
+            ;;
+    esac
+done
+
```
Comment (lguohan) [inline dockers/docker-fpm-frr/bgpd.sh:14]: you need shift $((OPTIND-1)

File: dockers/docker-fpm-frr/bgpd.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+addr="127.0.0.1"
+port=2601
+
+while getopts ":A:P:" opt; do
+    case "${opt}" in
+        A) addr=${OPTARG}
```
Comment (lguohan) [inline dockers/docker-fpm-frr/bgpd.sh:8]: this option conflicts with bgpd option, please use another option that does not conflict. please check the option bgpd has.

File: dockers/docker-fpm-frr/bgpd.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+            ;;
+        P) port=${OPTARG}
+            ;;
+    esac
+done
+
+timeout 5s bash -c -- "until </dev/tcp/${addr}/${port}; do sleep 0.1;done"
```
Comment (lguohan) [inline dockers/docker-fpm-frr/bgpd.sh:15]: what if timeout? I think we should print error msg?

File: dockers/docker-fpm-frr/bgpd.sh (hunk 1 above)
Comment (shi-su) [reply dockers/docker-fpm-frr/bgpd.sh:27]: Thanks for this comment, added the shift.

File: dockers/docker-fpm-frr/bgpd.sh (hunk 2 above)
Comment (shi-su) [reply dockers/docker-fpm-frr/bgpd.sh]: Updated the options not to conflict with bgpd

File: dockers/docker-fpm-frr/bgpd.sh (hunk 3 above)
Comment (shi-su) [reply dockers/docker-fpm-frr/bgpd.sh:28]: Added a check for exit code and log a message if failed.

File: dockers/docker-fpm-frr/bgpd.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            ;;
+        o) port=${OPTARG}
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+timeout 5s bash -c -- "until </dev/tcp/${addr}/${port}; do sleep 0.1;done"
+if [ "$?" != "0" ]; then
+    logger "Error: zebra is not ready to accept connections"
```
Comment (lguohan) [inline dockers/docker-fpm-frr/bgpd.sh:30]: logger -p error to indicate error.

Comment (shi-su) [reply dockers/docker-fpm-frr/bgpd.sh]: Thanks for this suggestion, added a priority option to indicate error.

**Review by lguohan — APPROVED** [review]

**Comment by shi-su** [comment]
retest vsimage please

**Comment by lguohan** [comment]
azure pipeline kvmtest passed.

**Pull request merged**